django/db/models/functions/text.py

```diff
@@ -188,6 +191,125 @@ def as_sqlite(self, compiler, connection, **extra_context):
+
+    @cached_property
+    def flag_mapping(self):
+        return {k: str.maketrans(v) for k, v in {
+            'mariadb': {'c': '-i'},
+            'mysql': {'s': 'n'},
+            'oracle': {'s': 'n'},
+            'postgresql': {'m': 'n'},
+        }.items()}
+
+    def as_sql(self, compiler, connection, **extra_context):
+        clone = self.copy()
+        expression, pattern, *extra, flags = clone.get_source_expressions()
+
+        if connection.vendor == 'mysql' and connection.mysql_is_mariadb:
+            vendor = 'mariadb'
+        else:
+            vendor = connection.vendor
+
+        # Add default flags to ensure consistency between database backends.
+        if vendor in self.default_flags:
+            flags.value = self.default_flags[vendor] + flags.value
+
+        # Wrap pattern in group if required and increment backreferences.
+        if pattern.value and vendor in self.wrap_pattern and not extra_context.get('no_wrap'):
+            pattern.value = '(%s)' % re.sub(
+                r'\\([0-9])',
+                lambda m: r'\%d' % (int(m.group(1)) + 1),
+                pattern.value,
+            )
+
+        # Remove duplicate flags preserving the last occurrence of each to
+        # ensures that the last flag is preferred if any are contradictory.
+        # PY38: list() is no longer required to use reversed() on dict().
+        if flags and flags.value:
+            flags.value = ''.join(reversed(list(dict.fromkeys(reversed(flags.value)))))
+
+        # Resolve case-sensitivity flags preferring the last flag specified.
+        if {'c', 'i'}.issubset(flags.value):
+            remove = 'i' if flags.value.index('c') > flags.value.index('i') else 'c'
+            flags.value = flags.value.replace(remove, '')
+
+        if vendor in self.flag_mapping:
+            flags.value = flags.value.translate(self.flag_mapping[vendor])
+
+        if vendor in {'mysql', 'oracle'}:
+            extra.append(Value(1))  # position
+            extra.append(Value(0 if 'g' in flags.value else 1))  # occurrence
+            if isinstance(self, RegexpStrIndex):
+                extra.append(Value(0))  # return_opt
+            flags.value = flags.value.replace('g', '')
+
+        if vendor in self.inline_flags:
+            if flags.value and pattern.value:
+                # FIXME: Can this hack be improved?
+                if '-i' in flags.value:
+                    pattern.value = '(?-i)' + pattern.value
+                    flags.value = flags.value.replace('-i', '')
+                pattern.value = ('(?%s)' % flags.value.replace('g', '')) + pattern.value
+        else:
+            extra.append(flags)
+
+        clone.set_source_expressions([expression, pattern, *extra])
+        return super(RegexpFlagMixin, clone).as_sql(compiler, connection, **extra_context)
+
+
+class RegexpReplace(RegexpFlagMixin, Func):
+    function = 'REGEXP_REPLACE'
+    inline_flags = {'mariadb'}
+    output_field = CharField()
```

Comment:
@felixxm I've got some test failures on Oracle and was wondering if you as an oracle on Oracle would understand.

It seems that the value returned is an instance of `cx_Oracle.LOB` and I have to cast that to `str` to get a sensible value.

If I change this line to `TextField` the tests pass, but that differs to all of the other functions in this module which prefer `CharField`. I didn't have these failures originally and think that they may have appeared after 1e38f1191de21b6e96736f58df57dfb851a28c1f and I probably had to add this `output_field` line here after that due to mixed `CharField` and `TextField`. Maybe @charettes will understand what is going on here.

Reply:
I don't have a quick answer. I would expect that all text-functions are affected by this issue (but I couldn't reproduce it with `LPad` or `Left` :thinking:). Oracle should return `LOB`\`s output for `CLOB`\`s inputs, when we set `output_field` to `CharField()` it's not converted to `str` in contrast to `TextField`, see 

https://github.com/django/django/blob/35b03788b0607c1f8d2b64e4fa9e1669b0907ea4/django/db/backends/oracle/operations.py#L205-L208

Adding conversion will fix tests, but I'm not sure if that's a good approach:

```diff
diff --git a/django/db/backends/oracle/operations.py b/django/db/backends/oracle/operations.py
index 1e5dc70613..bd15c0a140 100644
--- a/django/db/backends/oracle/operations.py
+++ b/django/db/backends/oracle/operations.py
@@ -176,7 +176,7 @@ END;
     def get_db_converters(self, expression):
         converters = super().get_db_converters(expression)
         internal_type = expression.output_field.get_internal_type()
-        if internal_type in ['JSONField', 'TextField']:
+        if internal_type in ['JSONField', 'TextField', 'CharField']:
             converters.append(self.convert_textfield_value)
         elif internal_type == 'BinaryField':
             converters.append(self.convert_binaryfield_value)
```